drivers/net/ethernet/baikal/xgbe/xgbe-drv.c

```diff
@@ -384,84 +370,105 @@ static irqreturn_t xgbe_isr(int irq, void *data)
 	return IRQ_HANDLED;
 }
 
-static irqreturn_t xgbe_dma_tx_isr(int irq, void *data) {
+static irqreturn_t xgbe_dma_isr(int irq, void *data) {
 
 	struct xgbe_channel *channel = data;
 	unsigned int dma_ch_isr;
 
-	/* Per channel DMA interrupts are enabled, so we use the per
-	 * channel napi structure and not the private data napi structure
-	 */
-	if (napi_schedule_prep(&channel->napi_tx)) {
-		//pr_err("xgbe_dma_tx_isr NAPI\n");
-		/* Disable Tx interrupts */
-		disable_irq_nosync(channel->tx_dma_irq);
-
-                dma_ch_isr = XGMAC_DMA_IOREAD(channel, DMA_CH_SR);
-		/* Turn on polling */
-		__napi_schedule(&channel->napi_tx);
-
-		XGMAC_DMA_IOWRITE(channel, DMA_CH_SR, dma_ch_isr);
-	}
-
-        return IRQ_HANDLED;
-}
+    dma_ch_isr = XGMAC_DMA_IOREAD(channel, DMA_CH_SR);
 
-static irqreturn_t xgbe_dma_rx_isr(int irq, void *data) {
-
-	struct xgbe_channel *channel = data;
-	unsigned int dma_ch_isr;
-
-	/* Per channel DMA interrupts are enabled, so we use the per
-	 * channel napi structure and not the private data napi structure
-	 */
-	if (napi_schedule_prep(&channel->napi_rx)) {
-		/* Disable Rx interrupts */
+	if (napi_schedule_prep(&channel->napi)) {
 		disable_irq_nosync(channel->rx_dma_irq);
```

Comment:
При такой реализации следует еще добавить строку:
```
disable_irq_nosync(channel->tx_dma_irq);
```

Reply:
Не совсем понял зачем следует выполнить disable_irq_nosync(channel->tx_dma_irq);. xgbe_dma_isr - Top Hlaf обработчик. Для rx_dma_irq мы используем napi, поэтому выключаем прерывание. Если добавить отключение прерывания по tx, то при выходе из BH получим отключенное прерывание. По идее прерывания по tx можно вообще не отключать.

Reply:
> Запустил еще раз получил новый лог. Выкладываю для ознакомления.
> 
> ```
> BUG: Bad page state in process kpktgend_0  pfn:01fc8
> page:8205f820 count:-1 mapcount:0 mapping:  (null) index:0x0
> flags: 0x40000000()
> page dumped because: nonzero _count

На самом деле, похожих слетов очень много. И по разным причинам. Можно попробовать поиграться признаком dirty выделенных страницы, но пока ясного и простого решения у меня нет. Обдумываю варианты.